src/py/mat3ra/notebooks_utils/jupyterlite/url.py

```
@@ -0,0 +1,65 @@
+    Fetch, read, and decode content from a URL in a Python environment.
+
+    Args:
+        url (str): The URL to fetch from.
+        as_bytes (bool): Whether to return the content as bytes.
+
+    Returns:
+        str: The content as a string or bytes.
+    """
+    import urllib.request
+
+    with urllib.request.urlopen(url) as response:
```

Ruff (0.15.12)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)